### ISSUE TITLE
⚠️ NOTICE: a new major release of DeepLabCut, v3.0.0 is pending -- we need your feedback on v3.0.0rc1

### ISSUE THREAD

**Issue by MMathisLab** [DeepLabCut/DeepLabCut#2616]
## Excitingly, we are alpha releasing a full PyTorch DeepLabCut!

Any user, who has installed DeepLabCut since 2021, has PyTorch installed: we added PyTorch for unsupervised reID and tracking of animals ([Figure 4 in Lauer et al. 2022 nature methods](https://www.nature.com/articles/s41592-022-01443-0#Fig4)), so you just want to **update** to now use **the new source code** -- we have been preparing for this major, powerful update! 🔥💜 

## We want your input and feedback, so please comment below if you try it, what you love, what you want to see improved, and how it works for you!

- Please see the [updated Documentation](https://github.com/DeepLabCut/DeepLabCut/pull/2605)
- Please see the performance, source code, the authors, and **how to install v3.0.0rc1** in this [open PR](https://github.com/DeepLabCut/DeepLabCut/pull/2613)!

**Comment by Vaccummer** [comment]
I tried, but I can't install it. There is an error when I conduct 
"pip install "git+https://github.com/DeepLabCut/DeepLabCut.git@pytorch_dlc#egg=deeplabcut[gui,modelzoo,wandb]"
error: RPC failed; curl 56 GnuTLS recv error (-9): Error decoding the received TLS packet.

**Comment by MMathisLab** [comment]
Hi @Vaccummer thanks for testing! Maybe its an internet issue.... can you try 

`pip install "deeplabcut[gui,modelzoo,wandb]==3.0.0rc1"`

https://pypi.org/project/deeplabcut/3.0.0rc1/

**Comment by mikeumo** [comment]
Several comments:
1) somewhat annoying dialogue "DeepLabCut 2.3.10 available: Skip/Update" every time you start DLC 3.0

2) when I choose 2 cameras the video selection pane and "Browse folders" button get immediately greyed out and I cannot choose any file

3) with 1 camera selected I can browse the folders but cannot select any  AVI or MP4 file in the file selection dialog. When I click on a video, the "Choose" button gets greyed out. I can play the videos in VLC player, not QuickTime. The videos are located in Dropbox-managed folder.

So, how can I select the videos?

**Comment by MMathisLab** [comment]
Hi @mikeumo 

- the message pops up because you're not using a stable release; obviously will "go away" when you install stable, not during this test-only period.

- you need to select the folder, then all videos load as options. (ie.. others and we have success here ;) https://github.com/DeepLabCut/DeepLabCut/issues/2620)

- 2 cameras we don't support in alpha mode yet, hence why it's greyed out.

**Comment by BBrianZhang** [comment]
Hello, I'm pleased to use the Deeplabcut 3.x version. I would like to ask how to use the BUTCD model. Looking forward to your response.

**Comment by neuromancer32** [comment]
Encountered this error when building the conda environment using the provided .yaml file on Anaconda and Windows 10:
```
Pip subprocess error:
DEPRECATION: git+https://github.com/DeepLabCut/DeepLabCut.git@pytorch_dlc#egg=deeplabcut[gui,modelzoo,wandb] contains an egg fragment with a non-PEP 508 name pip 25.0 will enforce this behaviour change. A possible replacement is to use the req @ url syntax, and remove the egg fragment. Discussion can be found at https://github.com/pypa/pip/issues/11617
  ERROR: Error [WinError 2] The system cannot find the file specified while executing command git version
ERROR: Cannot find command 'git' - do you have 'git' installed and in your PATH?

failed

CondaEnvException: Pip failed
```
Adding ``` -git ``` line to the dependencies in the .yaml file appears to successfully build the environment

**Comment by pbentkowski** [comment]
Installing DLC 3.0 rc1 and running it with PyTorch was SO MUCH easier than 2.x series with Tensorflow! However, I get a message:
```
Cannot run videos analysis for top-down models without a detector snapshot. Please specify your desired detector_snapshotindex in your project's configuration file.
```
But the computation continues, continues, conti… for several hours now. Yes, I picked a top-down model as we filmed the animals with a camera hanging over their heads. What kind of data I need to provide?

**Comment by Vaccummer** [comment]
1. if Task name or Scorer name has "." , program will produce a wrong training dataset path due to function ".with_suffix"
2. huggingface hub is absolutely blocked in China, and I can't mannually download cause I dont know where it's saved.
3. I tried hrnet, but encountered a performance problem. The "head.target_generator" consume much more time than backbone inference and backwards. I assume this function is conducted on cpu rather than gpu.

**Comment by Vaccummer** [comment]
> Hi @Vaccummer thanks for testing! Maybe its an internet issue.... can you try
> 
> `pip install "deeplabcut[gui,modelzoo,wandb]==3.0.0rc1"`
> 
> https://pypi.org/project/deeplabcut/3.0.0rc1/

thanks， it worked， but there are three more probelms:
1. if Task name or Scorer name has "." , program will produce a wrong training dataset path due to function ".with_suffix"
2. huggingface hub is absolutely blocked in China, and I can't mannually download init_weight cause I dont know where it's saved.
3. I tried hrnet, but encountered a performance problem. The "head.target_generator" consume much more time than backbone inference and backwards. I assume this function is conducted on cpu rather than gpu.

**Comment by BBrianZhang** [comment]
> > Hi @Vaccummer thanks for testing! Maybe its an internet issue.... can you try
> > `pip install "deeplabcut[gui,modelzoo,wandb]==3.0.0rc1"`
> > https://pypi.org/project/deeplabcut/3.0.0rc1/
> 
> thanks， it worked， but there are three more probelms:
> 
> 1. if Task name or Scorer name has "." , program will produce a wrong training dataset path due to function ".with_suffix"
> 2. huggingface hub is absolutely blocked in China, and I can't mannually download init_weight cause I dont know where it's saved.
> 3. I tried hrnet, but encountered a performance problem. The "head.target_generator" consume much more time than backbone inference and backwards. I assume this function is conducted on cpu rather than gpu.

You can directly add
 `os.environ["HF_ENDPOINT"] = "https://hf-mirror.com"`
after 
`import os` in `site-packages\huggingface_hub\__init__.py`. 
https://hf-mirror.com is a domestic mirror source.

**Comment by Mijar007** [comment]
Hi,
I found two small bugs in the GUI (Win11 PC).
1.) If you create a project (videos copied to project folder), the app creates also for every video a folder in the 'labeled-data' directory. This becomes annoying if you only extract frames from a subset of the videos and analyze another subset. If you choose the 'Extract outlier frames' path afterwards, you have to manually delete all empty folder before merging. I think it is better if the 'labeled-data' folders are created based on the 'Extract frames' or outlier selection.
2.) In the 'Create video' section, the 'Overwrite videos' checkbox doesn't work. If you have it selected and create videos, already created ones won't be replaced. You have to manually delete the already created videos if you want to recreate them with different settings.

**Comment by MMathisLab** [comment]
> Hi,
> 
> I found two small bugs in the GUI (Win11 PC).
> 
> 1.) If you create a project (videos copied to project folder), the app creates also for every video a folder in the 'labeled-data' directory. This becomes annoying if you only extract frames from a subset of the videos and analyze another subset. If you choose the 'Extract outlier frames' path afterwards, you have to manually delete all empty folder before merging. I think it is better if the 'labeled-data' folders are created based on the 'Extract frames' or outlier selection.
> 
> 2.) In the 'Create video' section, the 'Overwrite videos' checkbox doesn't work. If you have it selected and create videos, already created ones won't be replaced. You have to manually delete the already created videos if you want to recreate them with different settings.

#1 is the expected behavior and did not change in this PR; we always keep images from different videos in separate folders. The rest I cannot understand- please make a new issue on GitHub vs posting here so we can better assist. 

#2 @jeylau please check

**Comment by BBrianZhang** [comment]
Hello, 
while using Deeplabcut with a **Pytorch** engine, I encountered an issue with the model **dlcrnet_stride16_ms5**. During the training process, a ValueError occurred stating "**matrix contains invalid numeric entries.**" No matter how much I reduce the learning rate or adjust the training batch size, I cannot resolve this problem.

**Comment by MMathisLab** [comment]
> 
> 
> 
> 
> 
> 
> > Hi @Vaccummer thanks for testing! Maybe its an internet issue.... can you try
> 
> > 
> 
> > `pip install "deeplabcut[gui,modelzoo,wandb]==3.0.0rc1"`
> 
> > 
> 
> > https://pypi.org/project/deeplabcut/3.0.0rc1/
> 
> 
> 
> thanks， it worked， but there are three more probelms:
> 
> 1. if Task name or Scorer name has "." , program will produce a wrong training dataset path due to function ".with_suffix"
> 
> 2. huggingface hub is absolutely blocked in China, and I can't mannually download init_weight cause I dont know where it's saved.
> 
> 3. I tried hrnet, but encountered a performance problem. The "head.target_generator" consume much more time than backbone inference and backwards. I assume this function is conducted on cpu rather than gpu.

#1, correct - as we say in docs you cannot use special characters in your naming, esp "." Is a huge issue - imagine how many paths we need to parse, we need to use the . For splitting files ;) 

#3 you'd need to give more details and the trace back. Please create new issue vs posting comments here so we can better assist. @n-poulsen

**Comment by MMathisLab** [comment]
> Hello, @BBrianZhang 
> 
> while using Deeplabcut with a **Pytorch** engine, I encountered an issue with the model **dlcrnet_stride16_ms5**. During the training process, a ValueError occurred stating "**matrix contains invalid numeric entries.**" No matter how much I reduce the learning rate or adjust the training batch size, I cannot resolve this problem.

Please create a new issue with full details as required by our bug report 🙏🏼👍🚀

**Comment by BBrianZhang** [comment]
> > Hello, @BBrianZhang
> > while using Deeplabcut with a **Pytorch** engine, I encountered an issue with the model **dlcrnet_stride16_ms5**. During the training process, a ValueError occurred stating "**matrix contains invalid numeric entries.**" No matter how much I reduce the learning rate or adjust the training batch size, I cannot resolve this problem.
> 
> Please create a new issue with full details as required by our bug report 🙏🏼👍🚀


Thank you for your response. I have already created a new issue. #2631

**Comment by Vaccummer** [comment]
> > > Hi @Vaccummer thanks for testing! Maybe its an internet issue.... can you try
> > 
> > 
> > > 
> > 
> > 
> > > `pip install "deeplabcut[gui,modelzoo,wandb]==3.0.0rc1"`
> > 
> > 
> > > 
> > 
> > 
> > > https://pypi.org/project/deeplabcut/3.0.0rc1/
> > 
> > 
> > thanks， it worked， but there are three more probelms:
> > 
> > 1. if Task name or Scorer name has "." , program will produce a wrong training dataset path due to function ".with_suffix"
> > 2. huggingface hub is absolutely blocked in China, and I can't mannually download init_weight cause I dont know where it's saved.
> > 3. I tried hrnet, but encountered a performance problem. The "head.target_generator" consume much more time than backbone inference and backwards. I assume this function is conducted on cpu rather than gpu.
> 
> #1, correct - as we say in docs you cannot use special characters in your naming, esp "." Is a huge issue - imagine how many paths we need to parse, we need to use the . For splitting files ;)
> 
> #3 you'd need to give more details and the trace back. Please create new issue vs posting comments here so we can better assist. @n-poulsen

Thanks for reply, I will change my bad habit of naming^皿^
and I've submitted a issue about performance problem:
#2632

**Comment by zx377262372** [comment]
Python 3.11.4
Win11
When I run this "pip install "deeplabcut[gui,modelzoo,wandb]==3.0.0rc1", I got an error like this.
"""
ERROR: Could not find a version that satisfies the requirement npe2 (from versions: none)
ERROR: No matching distribution found for npe2
"""
please help me, thanks.

**Comment by jeylau** [comment]
@zx377262372, could you maybe retry with `pip install "git+https://github.com/DeepLabCut/DeepLabCut.git@pytorch_dlc#egg=deeplabcut[gui,modelzoo,wandb]`? I don't have a Windows machine to test `npe2` (which is napari's plugin engine) on, but at least it runs smoothly on my mac with python 3.11

**Comment by MMathisLab** [comment]
If you have issues, please create a new issue so we can easily track them in formatted bug reports, thanks everyone for contributing to making it even better!

**Comment by jeylau** [comment]
Hi @zx377262372, I encountered no installation issues on Windows with python 3.11. Perhaps you could open a new issue, and post a screenshot of your console and the whole traceback, we may find other clues in it to help you.

**Comment by AlexEMG** [comment]
Thanks for all the fixes & suggestions. Release candidate 3 is up, enjoy!

pip install "deeplabcut[gui,modelzoo,wandb]==3.0.0rc3"

https://pypi.org/project/deeplabcut/3.0.0rc3/

**Comment by MMathisLab** [comment]
3.0.0rc7 was merged into main! A FULL 3.0.0 is forthcoming.
